modules/nixos/default.nix

```diff
@@ -0,0 +1,114 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.options) mkOption;
+  inherit (lib.lists) filter map flatten concatLists;
+  inherit (lib.attrsets) filterAttrs mapAttrs' attrValues mapAttrsToList;
+  inherit (lib.trivial) flip;
+  inherit (lib.types) bool attrsOf submoduleWith listOf raw;
+
+  cfg = config.hjem;
+
+  hjemModule = submoduleWith {
+    description = "Hjem NixOS module";
+    class = "hjem";
+    specialArgs = {inherit pkgs lib;};
+    modules = concatLists [
+      [
+        ({name, ...}: {
+          imports = [../common.nix];
+
+          config = {
+            user = config.users.users.${name}.name;
+            directory = config.users.users.${name}.home;
+            clobberFiles = cfg.clobberByDefault;
+          };
+        })
+      ]
+
+      # Evaluate additional modules under 'hjem.users.<name>' so that
+      # module systems built on Hjem are more ergonomic.
+      cfg.extraModules
+    ];
+  };
+in {
+  imports = [
+    # This should be removed in the future, since 'homes' is a very vague
+    # namespace to occupy. Added 2024-12-27, remove 2025-01-27 to allow
+    # sufficient time to migrate.
+    (lib.mkRenamedOptionModule ["homes"] ["hjem" "users"])
+  ];
+
+  options.hjem = {
+    clobberByDefault = mkOption {
```

Comment:
any reason to not put this in the common file?

Reply:
This is bit of a hack. We cannot pass the `config` into the submodule directly, unless you want to do something like `specialArgs = {osConfig = config;}`  just to make sure NixOS config is directly accessible in the submodule. Thus we

1. define `clobberByDefault` in the NixOS module
2. default value of `clobberFiles` (in common.nix) gets set to this value
3. the value of `clobber` in the files submodule gets set to the value of `clobberFiles`

Not very clean, but the only way I could figure out how to propagate the top-level module option to a submodule inside a submodule.